### PR TITLE
Block version of before action

### PR DESCRIPTION
**Pull request by heysyam99**

### CODE REVIEW

File: lib/rage/controller/api.rb [hunk 1]
```diff
@@ -116,7 +116,19 @@ def rescue_from(*klasses, with: nil, &block)
...
+    #
+    #   before_action do
+    #     print("Who am i")
+    #   end
+    def before_action(action_name = nil, only: nil, except: nil, &block)
+      if block_given?
+        name = ('a'..'z').to_a.sample(15).join
+        action_name = define_method("__#{name}", &block)
+      elsif action_name.nil?
+        raise "No handler provided. Pass the `action_name` keyword argument or provide a block."
```
Comment (rsamoilov) [inline lib/rage/controller/api.rb:129]:
```suggestion
        raise "No handler provided. Pass the `action_name` parameter or provide a block."
```

File: spec/controller/api/before_actions_spec.rb [hunk 2]
```diff
@@ -106,6 +106,16 @@ def index
       verifier.setup_6
     end
   end
+
+  class TestController7 < RageController::API
+    before_action do
+      print("Hello Rage")
```
Comment (rsamoilov) [inline spec/controller/api/before_actions_spec.rb:112]:
```suggestion
      verifier.setup_7
```

File: spec/controller/api/before_actions_spec.rb [hunk 3]
```diff
@@ -213,4 +223,22 @@ def index
       expect(run_action(base_klass, :index)).to match([200, instance_of(Hash), ["hi from base"]])
     end
   end
+
+  context 'case 7' do
+    let(:klass) { ControllerApiBeforeActionsSpec::TestController7 }
+
+    it 'executes the block as a method and adds it to before actions' do
+      expect { run_action(klass, :index) }.to output("Hello Rage").to_stdout
```
Comment (rsamoilov) [inline spec/controller/api/before_actions_spec.rb:231]: That's actually interesting. I didn't know about the `output` helper.

However, do you want to update it to match the other tests? e.g. https://github.com/rage-rb/rage/blob/3a131858a2bf3ea02751d06189d40eb1b3e1150e/spec/controller/api/before_actions_spec.rb#L216-L219

**Review by rsamoilov — COMMENTED** [review]
Great job, Hisyam!

I've left some minor comments.

File: spec/controller/api/before_actions_spec.rb (hunk 3 above)
Comment (heysyam99) [reply spec/controller/api/before_actions_spec.rb:231]: Yeahh, I just realized the print statement will just pollute the test screen. Will change to match other tests

**Review by rsamoilov — APPROVED** [review]
Looks good! Thank you!

**Pull request merged**